drivers/counter/counter_ll_stm32_timer.c

```diff
@@ -0,0 +1,632 @@
+	init.ClockDivision = LL_TIM_CLOCKDIVISION_DIV1;
+
+	if (LL_TIM_Init(timer, &init) != SUCCESS) {
+		LOG_ERR("Could not initialize timer");
+		return -EIO;
+	}
+
+	return 0;
+}
+
+static uint32_t get_guard_period(const struct device *dev, uint32_t flags)
+{
+	return get_dev_data(dev)->guard_period;
+}
+
+static int set_guard_period(const struct device *dev, uint32_t guard,
+			    uint32_t flags)
+{
+	__ASSERT_NO_MSG(guard < get_top_value(dev));
+
+	get_dev_data(dev)->guard_period = guard;
+	return 0;
+}
+
+static void top_irq_handle(const struct device *dev)
+{
+	counter_top_callback_t cb = get_dev_data(dev)->top_cb;
+
+	__ASSERT(cb != NULL, "top event enabled - expecting callback");
+	cb(dev, get_dev_data(dev)->top_user_data);
+}
+
+static void alarm_irq_handle(const struct device *dev, uint32_t id)
+{
+	TIM_TypeDef *timer = get_stm32_config(dev)->timer;
+
+	struct counter_stm32_ch_data *chdata;
+	counter_alarm_callback_t cb;
+
+	atomic_and(&get_dev_data(dev)->cc_int_pending, ~BIT(id));
+	disable_it[id](timer);
+
+	chdata = &get_stm32_config(dev)->ch_data[id];
+	cb = chdata->callback;
+	chdata->callback = NULL;
+
+	if (cb) {
+		uint32_t cc_val = get_timer_compare[id](timer);
+
+		cb(dev, id, cc_val, chdata->user_data);
+	}
+}
+
+static const struct counter_driver_api counter_stm32_driver_api = {
+	.start = start,
+	.stop = stop,
+	.get_value = get_value,
+	.set_alarm = set_alarm,
+	.cancel_alarm = cancel_alarm,
+	.set_top_value = set_top_value,
+	.get_pending_int = get_pending_int,
+	.get_top_value = get_top_value,
+	.get_guard_period = get_guard_period,
+	.set_guard_period = set_guard_period,
+};
+
+
+#define TIM_IRQ_HANDLE_CC(timx, cc)						   \
+	do {									   \
+		bool hw_irq = LL_TIM_IsActiveFlag_CC##cc(timer) &&		   \
+			      LL_TIM_IsEnabledIT_CC##cc(timer);			   \
+		if (hw_irq || (get_dev_data(dev)->cc_int_pending & BIT(cc - 1))) { \
+			if (hw_irq) {						   \
+				LL_TIM_ClearFlag_CC##cc(timer);			   \
+			}							   \
+			alarm_irq_handle(dev, cc - 1);				   \
+		}								   \
+	} while (0)
+
+void tim_irq_handler(const struct device *dev)
+{
+	TIM_TypeDef *timer = get_stm32_config(dev)->timer;
+
+	/* Capture compare events */
+	switch (counter_get_num_of_channels(dev)) {
+	case 4:
+		TIM_IRQ_HANDLE_CC(timer, 4);
+	case 3:
+		TIM_IRQ_HANDLE_CC(timer, 3);
+	case 2:
+		TIM_IRQ_HANDLE_CC(timer, 2);
+	case 1:
+		TIM_IRQ_HANDLE_CC(timer, 1);
+	}
+
+	/* TIM Update event */
+	if (LL_TIM_IsActiveFlag_UPDATE(timer) && LL_TIM_IsEnabledIT_UPDATE(timer)) {
+		LL_TIM_ClearFlag_UPDATE(timer);
+		top_irq_handle(dev);
+	}
+}
+
+#define TIMER(idx)              DT_PARENT(DT_DRV_INST(idx))
+
+#define TIMER_IRQ_CONNECT(idx)				 \
+	IRQ_DIRECT_CONNECT(DT_IRQN(TIMER(idx)),		 \
+			   DT_IRQ(TIMER(idx), priority), \
+			   counter##idx##_irq_handler,	 \
+			   IRQ_ZERO_LATENCY)
+
+/** TIMx instance from DT */
+#define TIM(idx) ((TIM_TypeDef *)DT_REG_ADDR(DT_PARENT(DT_DRV_INST(idx))))
+
+#define DT_INST_CLK(idx)				\
```

Comment:
DT_INST_ namespace is reserved for generic usage. Use TIMER_CLK for instance

Reply:
Ok, changed.